src/session/session_api.c

```diff
@@ -488,6 +488,27 @@ __session_reconfigure(WT_SESSION *wt_session, const char *config)
         session->cache_max_wait_us = (uint64_t)(cval.val * WT_THOUSAND);
     WT_ERR_NOTFOUND_OK(ret, false);
 
+    if (S2C(session)->prefetch_auto_on)
+        F_SET(session, WT_SESSION_PREFETCH);
+    else
+        F_CLR(session, WT_SESSION_PREFETCH);
+
+    /*
+     * Override any connection-level pre-fetch settings if a specific session-level setting was
+     * provided.
+     */
+    WT_ERR(__wt_config_gets(session, cfg, "prefetch.enabled", &cval));
+    if (cval.val) {
+        if (!S2C(session)->prefetch_available) {
+            F_CLR(session, WT_SESSION_PREFETCH);
+            WT_ERR_MSG(session, EINVAL,
+              "pre-fetching cannot be enabled for the session if pre-fetching server is "
+              "unavailable");
+        } else
+            F_SET(session, WT_SESSION_PREFETCH);
+    } else
+        F_CLR(session, WT_SESSION_PREFETCH);
```

Comment:
Won't this force prefetching off if the session-level prefetch config is left blank? Or am I misunderstanding something here?

Reply:
Good catch, it will force the session-level prefetch config which isn't necessarily what we want. Still thinking about the best way around this - maybe we'll have to move from a boolean to an on/off string so that we can explicitly check for the string values instead?

Reply:
That's one option. I don't remember what else hangs off `cval` - is it just `val`, or are there other fields to detect if it was explicitly set?

Reply:
I don't see any other fields that can be used to determine if it was explicitly set.

Reply:
I agree with @mm-ng , I don't see any other fields that can be used. I think moving to a string value might be a good approach here.

Reply:
Fair enough - thanks for taking a look.

Reply:
Got a solution :-) I'll push it in the next commit.